### PR TITLE
Added tests, updated doc-strings for Dummy Datasets

### PR DESCRIPTION
**Pull request by nishantb06**
## What does this PR do?

Part of  #839 

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # IDE Settings files
 .vscode/
 
+
```
Comment (otaj) [inline .gitignore:4]: Can we not have even these changes?

File: tests/datasets/test_datasets.py [hunk 2]
```diff
@@ -1,41 +1,144 @@
...
 
+        batch = next(iter(dl))
+        assert batch[0].shape == torch.Size([batch_size, 1, 28, 28])
+        assert batch[1].shape == torch.Size([batch_size, 1])
 
-def test_rand_dict_ds():
-    ds = RandomDictDataset(32, num_samples=100)
-    dl = DataLoader(ds)
+    else:
+        with pytest.raises(Exception) as e:
```
Comment (otaj) [inline tests/datasets/test_datasets.py:30]: We can expect specific exception type (`ValueError`) and match the string of exception in this context manager

Comment (otaj) [reply tests/datasets/test_datasets.py:30]: This applies to all other tests

File: pl_bolts/datasets/dummy_dataset.py [hunk 3]
```diff
@@ -94,14 +105,25 @@ class RandomDictDataset(Dataset):
...
             num_samples: number of samples
         """
+        if num_samples < 1:
+            raise ValueError("Provide an argument greater than 0 for `num_samples`")
+
+        if size < 1:
+            raise ValueError("Provide an argument greater than 0 for `size`")
+
         self.len = num_samples
         self.data = torch.randn(num_samples, size)
```
Comment (otaj) [inline pl_bolts/datasets/dummy_dataset.py:128]: can `b` in the dictionary also be random and not dependant on `a`?

**Review by otaj — APPROVED** [review]
Hi, it looks good, thanks! I left couple comments :zap:

File: tests/datasets/test_datasets.py (hunk 2 above)
Comment (nishantb06) [reply tests/datasets/test_datasets.py:30]: @otaj . sorry I didn't understand what this means exactly? I mean I have done exactly that in the next two `assert` statements. Did you mean that the assert statements should be inside the context manager as well?

Comment (otaj) [reply tests/datasets/test_datasets.py:30]:
```suggestion
        with pytest.raises(ValueError, match="regexp to match the error text"):
```
and you can omit those two `asserts` then :rocket:

File: pl_bolts/datasets/dummy_dataset.py (hunk 3 above)
Comment (nishantb06) [reply pl_bolts/datasets/dummy_dataset.py]: made the commit with the necessary changes.thanks!

File: tests/datasets/test_datasets.py (hunk 2 above)
Comment (nishantb06) [reply tests/datasets/test_datasets.py]: made the commit with the necessary changes.thanks!

**Pull request merged**